src/frequenz/sdk/timeseries/_resampling.py

```diff
@@ -429,6 +429,30 @@ def remove_timeseries(self, source: Source) -> bool:
             return False
         return True
 
+    async def _wait_for_next_resampling_period(self) -> None:
+        """Wait for next resampling period.
+
+        If resampling period already started, then return without sleeping.
+        That would allow us to catch up with resampling.
+        Print warning if function woke up to late.
+        """
+        now = datetime.now(tz=timezone.utc)
+        if self._window_end > now:
+            sleep_for = self._window_end - now
+            await asyncio.sleep(sleep_for.total_seconds())
+
+        timer_error_s = (now - self._window_end).total_seconds()
+        if timer_error_s > (self._config.resampling_period_s / 10.0):
+            _logger.warning(
+                "The resampling started too late. It should have started at "
```

Comment:
I would still mention something more towards the timer drifting here, IMHO is less clear if we just say it *started too late*, why did it start too late?